engine/src/main/java/org/destinationsol/game/SolGame.java

```diff
@@ -106,73 +140,17 @@
-        solApplication = context.get(SolApplication.class);
-
-        //TODO this no longer needs to be instantiated in SolGame
-        GameColors gameColors = new GameColors();
-        context.put(GameColors.class, gameColors);
-
-        //TODO this no longer needs to be instantiated in SolGame
-        soundManager = solApplication.getSoundManager();
-        SpecialSounds specialSounds = new SpecialSounds(soundManager);
-        context.put(SpecialSounds.class, specialSounds);
-
-        //TODO these no longer need to be instantiated in SolGame
-        GameDrawer drawer = new GameDrawer(commonDrawer);
-        context.put(GameDrawer.class, drawer);
-        DrawableManager drawableManager = new DrawableManager(drawer);
-        context.put(DrawableManager.class, drawableManager);
-
-        //TODO refactor the usages of this to be independent from SolGame, so that this can be deleted
-        // The only methods still using this are those that require changing an interface signature (UpdateAwareSystem, SolObject, and Drawable)
-        SolCam camera = context.get(SolCam.class);
-        solCamReference = camera;
-
-        gameScreens = new GameScreens(solApplication, context);
-
-        if (isTutorial) {
-            tutorialManager = new TutorialManager(gameScreens, solApplication.isMobile(), solApplication.getOptions(), this);
-            context.put(TutorialManager.class, tutorialManager);
-        } else {
-            tutorialManager = null;
-        }
-        farBackgroundManagerOld = new FarBackgroundManagerOld();
-        shipBuilder = new ShipBuilder();
-        EffectTypes effectTypes = new EffectTypes();
-        specialEffects = new SpecialEffects(effectTypes, gameColors);
-        itemManager = new ItemManager(soundManager, effectTypes, gameColors);
-        AbilityCommonConfigs abilityCommonConfigs = new AbilityCommonConfigs(effectTypes, gameColors, soundManager);
-        hullConfigManager = new HullConfigManager(itemManager, abilityCommonConfigs);
-        planetManager = new PlanetManager(hullConfigManager, gameColors, itemManager);
-        contactListener = new SolContactListener(this);
-        factionManager = new FactionManager();
-        objectManager = new ObjectManager(contactListener, factionManager, context);
-        context.put(World.class, objectManager.getWorld());
-        gridDrawer = new GridDrawer();
-        chunkManager = new ChunkManager();
-        partMan = new PartMan();
-        asteroidBuilder = new AsteroidBuilder();
-        lootBuilder = new LootBuilder();
-        mapDrawer = new MapDrawer();
-        rubbleBuilder = new RubbleBuilder();
-        context.put(RubbleBuilder.class, rubbleBuilder);
-        galaxyFiller = new GalaxyFiller(hullConfigManager);
-        starPortBuilder = new StarPort.Builder();
-        drawableDebugger = new DrawableDebugger();
-        mountDetectDrawer = new MountDetectDrawer();
-        beaconHandler = new BeaconHandler();
```

Comment:
all this has been moved into a single ServiceRegistry. might make sense to split this into different registries for testable. i.e constructing a world generator.